server/src/MeiernServer.ts

```diff
@@ -1,85 +1,172 @@
+    private activePlayer: ServerPlayer | undefined;
+    private lieValue = '';
+
+    constructor(server: Server) {
+        server.on("connection", (socket: Socket) => {
+            socket.on(SetPlayerNameAction.type, (setPlayerName) =>
+                this.addPlayer(setPlayerName, socket)
+            );
+            socket.on("disconnect", (reason: string) => {
+                this.handleDisconnect(socket, reason);
+            });
+            socket.on(GetPlayerListAction.type, () => {
+                socket.emit(
+                    SetPlayerListAction.type,
+                    new SetPlayerListAction(
+                        this.allPlayer.map((player) => player.toPlayer())
+                    )
+                );
+            });
+
+            socket.on(GetRandomDiceAction.type, () => {
+                this.globalDice = Dice.createRandomDice();
+                socket.emit(
+                    SetRandomDiceAction.type,
+                    new SetRandomDiceAction(this.globalDice)
+                );
+                if (this.globalDice.getDiceValue() === '21') {
+                    socket.emit(
+                        RevealDiceAction.type,
+                        new RevealDiceAction(this.globalDice)
+                    );
+                    socket.broadcast.emit(
+                        RevealDiceAction.type,
+                        new RevealDiceAction(this.globalDice)
+                    );
+                }
+            });
+            socket.on(RevealDiceAction.type, () => {
+                if (this.globalDice !== undefined) {
+                // Should work with 1 emit if we use io.emit but it wont work right now so this work around
+                    socket.emit(
+                        RevealDiceAction.type,
+                        new RevealDiceAction(this.globalDice)
+                    );
+                    socket.broadcast.emit(
+                        RevealDiceAction.type,
+                        new RevealDiceAction(this.globalDice)
+                    );
+                }
+
+            });
+
+            socket.on('getPlayerList', () => {
+
+                socket.emit(
+                    'getPlayerList',
+                    this.allPlayer
+                );
+
+            });
+            socket.on('reset', () => {
+             this.lieValue = '';
+            this.id = 0;
+            this.allPlayer = [];
+            this.globalDice = undefined;
+            this.activePlayer = undefined;
+                this.broadcastPlayerList(socket);
+            });
+            socket.on('startGame', () => {
+                 socket.emit('playersXTurn',
+                     this.allPlayer[0], '');
+                  this.activePlayer = this.allPlayer[0];
+            });
+            socket.on('getMePlayer', (id: string) => {
+               for (let player of this.allPlayer) {
+                   if (player.socketId == id) {
+                       socket.emit('setMePlayer', player);
```

Comment:
```suggestion
                       socket.emit('setMePlayer', player);
                       break;
```
Du hast dein Player danach musst du nicht mehr weiter suchen.

Reply:
klar :)